src/platform/icelake/include/platform/alh.h

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2019 Intel Corporation. All rights reserved.
+//
+// Author: Slawomir Blauciak <slawomir.blauciak@linux.intel.com>
+
+#ifndef SOF_ALH_H
```

Comment:
Your solution with undef is even better than my initial idea, but could we follow already implemented pattern? See e.g. include/platform/lib/clk.h. I would like to be consistent across the whole source code, so:
```
#if SOF_DRIVERS_ALH_H

#ifndef __PLATFORM_DRIVERS_ALH_H__
#define __PLATFORM_DRIVERS_ALH_H__

<code here>

#endif /* __PLATFORM_DRIVERS_ALH_H__ */

#else 

#error "This file shouldn't be included from outside of sof/drivers/alh.h"

#endif /* SOF_DRIVERS_ALH_H */
```